tests/unit/test_resolver.py

```diff
@@ -242,3 +271,66 @@ def test_resolve_with_multiple_stable_package_managers(tmp_path: Path) -> None:
 
         mock_resolve_gomod.assert_has_calls([mock.call(request)])
         mock_resolve_pip.assert_has_calls([mock.call(request)])
+
+
+@pytest.mark.parametrize(
+    "experimental_type,expected_base_type",
+    [
+        ("x-shrubbery", "shrubbery"),
+        ("x-coconut", "coconut"),
+    ],
+)
+def test_x_prefix_package_managers_are_processed_correctly_when_enabled(
+    experimental_type: str, expected_base_type: str, tmp_path: Path
+) -> None:
+    """Test that x-pkg experimental types are converted to pkg and handled correctly alongside stable packages."""
+    mock_gomod_resolver = mock.Mock(return_value=RequestOutput.empty())
+    mock_experimental_resolver = mock.Mock(return_value=RequestOutput.empty())
+
+    with (
+        mock.patch.dict(
+            resolver._package_managers,
+            values={"gomod": mock_gomod_resolver},
+            clear=True,
+        ),
+        mock.patch.dict(
+            resolver._dev_package_managers,
+            values={expected_base_type: mock_experimental_resolver},
+            clear=True,
+        ),
+        mock.patch(
+            "hermeto.core.models.input.get_args",
+            return_value=(
```

Comment:
The same question as before: do you need all these values? You are mocking a function anyway, might want to return just whatever is necessary for a test.